File: food-restaurant/src/main/kotlin/com/food/restaurant/domain/Restaurant.kt
package com.food.restaurant.domain

import java.util.*

data class Restaurant(
        val id: Long?,
        val uuid: UUID?,
        val name: String,
        val category: Category,
        val logo: String,
        val description: String,
        val address: String,
        val itens: List<MenuItem>
) {
    constructor(name: String, category: Category, logo: String,
                description: String, address: String, itens: List<MenuItem>) :
            this(null, null, name, category, logo, description, address, itens)
}

File: food-ui-webapp/src/domain/Restaurant.ts
export default class Restaurant {
  _id: string;
  _name: string;
  _logo: string;
  _description: string;
  _address: string;

  constructor(id: string, name: string, logo: string, description: string, 
              address: string) {
    this._id = id;
    this._name = name;
    this._logo = logo;
    this._description = description;
    this._address = address;
  }

  get id(): string {
    return this._id;
  }

  get name(): string {
    return this._name;
  }

  get logo(): string {
    return this._logo;
  }

  get description(): string {
    return this._description;
  }

  get address(): string {
    return this._address;
  }
}